File: HTTPShortcuts/app/src/main/java/ch/rmy/android/http_shortcuts/adapters/ShortcutAdapter.java
package ch.rmy.android.http_shortcuts.adapters;

import android.content.Context;

import ch.rmy.android.http_shortcuts.R;
import ch.rmy.android.http_shortcuts.realm.models.PendingExecution;
import ch.rmy.android.http_shortcuts.realm.models.Shortcut;
import io.realm.RealmChangeListener;
import io.realm.RealmResults;

public abstract class ShortcutAdapter extends BaseAdapter<Shortcut> {

    private final RealmChangeListener<RealmResults<PendingExecution>> changeListener = new RealmChangeListener<RealmResults<PendingExecution>>() {
        @Override
        public void onChange(RealmResults<PendingExecution> results) {
            notifyDataSetChanged();
        }
    };
    RealmResults<PendingExecution> shortcutsPendingExecution;

    ShortcutAdapter(Context context) {
        super(context);
    }

    @Override
    protected int getEmptyMarkerStringResource() {
        return R.string.no_shortcuts;
    }

    public void setPendingShortcuts(RealmResults<PendingExecution> shortcutsPendingExecution) {
        if (this.shortcutsPendingExecution != null) {
            this.shortcutsPendingExecution.removeChangeListener(changeListener);
        }
        this.shortcutsPendingExecution = shortcutsPendingExecution;
        if (this.shortcutsPendingExecution != null) {
            this.shortcutsPendingExecution.addChangeListener(changeListener);
        }
    }

}

File: HTTPShortcuts/app/src/main/kotlin/ch/rmy/android/http_shortcuts/adapters/ShortcutAdapter.kt
package ch.rmy.android.http_shortcuts.adapters

import android.content.Context

import ch.rmy.android.http_shortcuts.R
import ch.rmy.android.http_shortcuts.extensions.color
import ch.rmy.android.http_shortcuts.realm.livedata.ListLiveData
import ch.rmy.android.http_shortcuts.realm.models.PendingExecution
import ch.rmy.android.http_shortcuts.realm.models.Shortcut

abstract class ShortcutAdapter internal constructor(context: Context, shortcuts: ListLiveData<Shortcut>) : BaseAdapter<Shortcut>(context, shortcuts) {

    var textColor: TextColor = TextColor.DARK
        set(value) {
            field = value
            notifyDataSetChanged()
        }

    internal var shortcutsPendingExecution: List<PendingExecution> = emptyList()

    override val emptyMarkerStringResource = R.string.no_shortcuts

    fun setPendingShortcuts(shortcutsPendingExecution: List<PendingExecution>) {
        this.shortcutsPendingExecution = shortcutsPendingExecution
        notifyDataSetChanged()
    }

    protected val nameTextColor
        get() = when (textColor) {
            TextColor.BRIGHT -> color(context, R.color.text_color_primary_bright)
            TextColor.DARK -> color(context, R.color.text_color_primary_dark)
        }

    protected val descriptionTextColor
        get() = when (textColor) {
            TextColor.BRIGHT -> color(context, R.color.text_color_secondary_bright)
            TextColor.DARK -> color(context, R.color.text_color_secondary_dark)
        }

    enum class TextColor {
        BRIGHT, DARK
    }

}
